ui/intermittent-failures/DropdownMenuItems.jsx

```diff
@@ -3,56 +3,30 @@ import Icon from 'react-fontawesome';
 import PropTypes from 'prop-types';
 import { DropdownMenu, DropdownItem } from 'reactstrap';
 
-export default class DropdownMenuItems extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      selectedItem: this.props.default,
-    };
-    this.changeSelection = this.changeSelection.bind(this);
-  }
-
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.default) {
-      this.setState({ selectedItem: nextProps.default });
-    }
-  }
-
-  changeSelection(event) {
-    const { selectedItem } = this.state;
-    const selectedText = event.target.innerText;
-
-    if (selectedText !== selectedItem) {
-      this.setState({ selectedItem: selectedText }, () => this.props.updateData(selectedText));
-    }
-  }
-
-  render() {
-    const { selectedItem } = this.state;
-    const { options } = this.props;
-
-    return (
-      <DropdownMenu>
-        {options.map(item =>
-          (<DropdownItem key={item} onClick={this.changeSelection}>
-            <Icon
-              name="check"
-              className={`pr-1 ${selectedItem === item ? '' : 'hide'}`}
-            />
-            {item}
-          </DropdownItem>))}
-      </DropdownMenu>
-    );
-  }
-}
+const DropdownMenuItems = ({ selectedItem, updateData, options }) =>
+(
+  <DropdownMenu>
+    {options.map(item =>
+      (<DropdownItem key={item} onClick={event => updateData(event.target.innerText)}>
+        <Icon
+          name="check"
+          className={`pr-1 ${selectedItem === item ? '' : 'hide'}`}
+        />
+        {item}
+      </DropdownItem>))}
+  </DropdownMenu>
+);
 
 DropdownMenuItems.propTypes = {
   updateData: PropTypes.func,
-  default: PropTypes.string,
+  selectedItem: PropTypes.string,
   options: PropTypes.arrayOf(PropTypes.string).isRequired,
 };
 
 DropdownMenuItems.defaultProps = {
   updateData: null,
-  default: null,
+  selectedItem: null,
 };
+
+export default DropdownMenuItems;
+
```

Comment:
stray newline (will be a lint failure with the new eslint)

Reply:
That was quick! :)

Reply:
Fixed.